src/analyses/base.ml

```diff
@@ -1142,7 +1142,9 @@ struct
       Invariant.none
 
   let query_invariant_global ctx g =
-    if GobConfig.get_bool "ana.base.invariant.enabled" then (
+    if GobConfig.get_bool "ana.base.invariant.enabled" && get_bool "exp.earlyglobs" then (
+      (* Currently these global invariants are only sound for single-threaded mode with earlyglobs. *)
```

Comment:
I think the way it is phrased is now a bit confusing. Maybe this is clearer?

```suggestion
      (* Currently these global invariants are only sound with earlyglobs enabled for both single- and multi-threaded programs. *) 
     (* Otherwise, the value of globals in single-threaded mode are not accounted for. *)
```